### PR TITLE
File provider not to return hidden files or file names starting with per...

### PR DESCRIPTION
**Pull request by Praburaj**
...iod

Fixes: https://github.com/aspnet/FileSystem/issues/51

@GrabYourPitchforks @glennc

### CODE REVIEW

File: src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs [hunk 1]
```diff
@@ -80,6 +80,14 @@ private void OnChanged(object sender, FileSystemEventArgs e)
 
         private void OnFileSystemEntryChange(string fullPath)
         {
+            var fileSystemInfo = new FileInfo(fullPath);
+            if (fileSystemInfo.Name.StartsWith(".") ||
+                (fileSystemInfo.Exists && fileSystemInfo.Attributes.HasFlag(FileAttributes.Hidden)))
```
Comment (Eilon) [inline src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs]: This looks like it should go in some static helper to re-use in the several places it's used.

Also, I think this needs to check for `FileAttributes.System`, which is a "super hidden" file.

Comment (Praburaj) [inline src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs]: sure. I can make it.

Comment (Praburaj) [inline src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs]: @Eilon I pushed an updated PR.

File: src/Microsoft.AspNet.FileProviders/Implementation/Helper.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+
+namespace Microsoft.AspNet.FileProviders
+{
+    public class Helper
```
Comment (Eilon) [inline src/Microsoft.AspNet.FileProviders/Implementation/Helper.cs]: Never name things `Helper`. Need a more specific name for this.

File: src/Microsoft.AspNet.FileProviders/Implementation/Helper.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        internal static bool CanBeServed(FileSystemInfo fileSystemInfo)
```
Comment (Eilon) [inline src/Microsoft.AspNet.FileProviders/Implementation/Helper.cs]: `CanBeServed` seems orthogonal to the use case. Isn't this just saying whether the file is hidden or not?

File: src/Microsoft.AspNet.FileProviders/Implementation/Helper.cs (hunk 2 above)
Comment (Praburaj) [inline src/Microsoft.AspNet.FileProviders/Implementation/Helper.cs]: How about `FileSystemInfoHelper` ?

File: src/Microsoft.AspNet.FileProviders/Implementation/Helper.cs (hunk 3 above)
Comment (Praburaj) [inline src/Microsoft.AspNet.FileProviders/Implementation/Helper.cs]: How about `IsReservedFile`? (Of course with return boolean negated).

Comment (Eilon) [inline src/Microsoft.AspNet.FileProviders/Implementation/Helper.cs]: They're not *reserved*, they're *hidden*. `IsHiddenFile` would be fine.

File: src/Microsoft.AspNet.FileProviders/Implementation/Helper.cs (hunk 2 above)
Comment (Eilon) [inline src/Microsoft.AspNet.FileProviders/Implementation/Helper.cs]: Yeah that seems fine.

File: src/Microsoft.AspNet.FileProviders/Implementation/Helper.cs (hunk 3 above)
Comment (Praburaj) [inline src/Microsoft.AspNet.FileProviders/Implementation/Helper.cs]: Cool. Will update it.

**Comment by Praburaj** [comment]
Updated PR.

**Comment by Eilon** [comment]
Looks :shipit: to me. @GrabYourPitchforks do you want to look to see if anything is missing?

**Comment by Praburaj** [comment]
@GrabYourPitchforks - do you have any comments?

**Comment by Praburaj** [comment]
Changes merged : https://github.com/aspnet/FileSystem/commit/0d2d9b1913175a764faaa57dc40de3e3761797c9
